iphone/cli/commands/_build.js

```diff
@@ -4891,1228 +4890,1175 @@ iOSBuilder.prototype.writeDebugProfilePlists = function writeDebugProfilePlists(
-							// iPad Portrait - iOS 7-9 - 2x (????)
-							'Default-Portrait@2x.png':       { idiom: 'ipad',   extent: 'full-screen', minSysVer: '7.0', orientation: 'portrait', width: 1536, height: 2048, scale: 2 },
-							// iPhone Portrait - iOS 8,9 - Retina HD 5.5 (1242x2208)
-							'Default-Portrait-736h@3x.png':  { idiom: 'iphone', extent: 'full-screen', minSysVer: '8.0', orientation: 'portrait', width: 1242, height: 2208, scale: 3, subtype: '736h' },
-							// iPhone Portrait - iOS 11 - Retina HD iPhone X (1125x2436)
-							'Default-Portrait-2436h@3x.png':  { idiom: 'iphone', extent: 'full-screen', minSysVer: '11.0', orientation: 'portrait', width: 1125, height: 2436, scale: 3, subtype: '2436h' },
-							// iPhone Portrait - iOS 12 - Retina HD iPhone X Max (1242x2688)
-							'Default-Portrait-2688h@3x.png':  { idiom: 'iphone', extent: 'full-screen', minSysVer: '12.0', orientation: 'portrait', width: 1242, height: 2688, scale: 3, subtype: '2688h' },
-							// iPhone Portrait - iOS 12 - Retina iPhone XR (828x1792)
-							'Default-Portrait-1792h@2x.png':  { idiom: 'iphone', extent: 'full-screen', minSysVer: '12.0', orientation: 'portrait', width: 828, height: 1792, scale: 2, subtype: '1792h' }
-						},
-						found = {};
 
-					fs.ensureDirSync(launchImageDir);
+							found[info.i18n || '_'] || (found[info.i18n || '_'] = {});
+							found[info.i18n || '_'][path.basename(filename)] = 1;
 
-					Object.keys(launchImages).forEach(function (filename) {
-						const info = launchImages[filename];
-						let meta = lookup[filename];
+							gatheredResults.resourcesToCopy.set(filename, info);
+						});
 
-						if (info.i18n) {
-							meta = lookup[path.basename(filename)];
-						}
+						// determine if we're missing any launch images
+						const missing = {};
+						let totalMissing = 0;
+						Object.keys(found).forEach(function (lang) {
+							Object.keys(lookup).forEach(function (filename) {
+								if (!found[lang][filename] && (this.deviceFamily !== 'ipad' || lookup[filename].idiom === 'ipad') && (this.deviceFamily !== 'iphone' || lookup[filename].idiom === 'iphone')) {
+									missing[lang] || (missing[lang] = {});
+									missing[lang][filename] = 1;
+									totalMissing++;
+								}
+							}, this);
+						}, this);
 
-						if (!meta) {
-							// we don't care about this image
-							this.logger.debug(__('Unsupported launch image %s, skipping', path.relative(this.projectDir, info.src).cyan));
-							return;
+						if (totalMissing) {
+							// we have missing launch images :(
+							this.logger.warn(__n('Missing a launch image:', 'Missing %s launch images:', totalMissing));
+							Object.keys(missing).forEach(function (lang) {
+								this.logger.warn('  ' + (lang === '_' ? __('Default') : 'i18n/' + lang));
+								Object.keys(missing[lang]).forEach(function (filename) {
+									const meta = lo
```

Comment:
<!--
  0 failure: 
  2 warning:  iphone/cli/comman..., iphone/cli/comman...
  
  
  DangerID: danger-id-Danger;
  File: iphone/cli/commands/_build.js;
  Line: 5966;
-->

- :warning: iphone/cli/commands/_build.js line 5966 – Avoid using promises inside of callbacks. (promise/no-promise-in-callback)
- :warning: iphone/cli/commands/_build.js line 5966 – Avoid using promises inside of callbacks. (promise/no-promise-in-callback)